back/dora/core/csv_user_import.py

```diff
@@ -0,0 +1,250 @@
+import csv
+import re
+from typing import Dict, List
+
+from dora.structures.emails import (
+    send_france_travail_invitation_email,
+    send_invitation_email,
+)
+from dora.structures.models import (
+    Structure,
+    StructureMember,
+    StructurePutativeMember,
+)
+from dora.users.models import User
+
+
+class ImportUserHelper:
+    def import_france_travail_users(
+        self,
+        reader: csv.reader,
+        wet_run: bool = False,
+        make_users_admin: bool = False,
+    ) -> None:
+        errors = []
+        users_to_import = []
+        self._display_run_type(wet_run)
+
+        [headers, *lines] = reader
+
+        if self._check_headers(headers, self.FRANCE_TRAVAIL_CSV_HEADERS):
+            return
+
+        lines = [dict(zip(headers, line)) for line in lines]
+        for index, row in enumerate(lines, 2):
+            try:
+                safir = row["safir"]
+                structure = self._structure_by_safir(safir)
+                email = row.get("email") or self._name_to_france_travail_email(
+                    row["prenom"], row["nom"]
+                )
+                first_name = row.get("prenom")
+                last_name = row.get("nom")
+
+                users_to_import.append(
+                    {
+                        "structure": structure,
+                        "email": email,
+                        "first_name": first_name,
+                        "last_name": last_name,
+                    }
+                )
+            except Exception as ex:
+                errors.append(f"[{index + 1}]: {ex}")
+
+        if errors:
+            print("Il y a des erreurs qui empêchent l'import :")
+            print(("\n".join(errors)))
+
+        else:
+            for index, user in enumerate(users_to_import):
+                print(
+                    f"{index}. Import {user['first_name']} {user['last_name']} pour l'agence avec le code SAFIR : {safir}"
+                )
+                if wet_run:
+                    self._import_user(
+                        user,
+                        is_france_travail=True,
+                        admin=make_users_admin,
+                    )
```

Comment:
`return` dans le `if`, suppression du `else` et dés-indentation. Comme ça on est sûr qu'on s'arrête après les erreurs.

```suggestion
            return

        for index, user in enumerate(users_to_import):
            print(
                f"{index}. Import {user['first_name']} {user['last_name']} pour l'agence avec le code SAFIR : {safir}"
            )
            if wet_run:
                self._import_user(
                    user,
                    is_france_travail=True,
                    admin=make_users_admin,
                )
```